### PR TITLE
update artifact-engine version

### PR DESCRIPTION
**Pull request by omeshp**

### CODE REVIEW

File: Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts [hunk 1]
```diff
@@ -2,9 +2,13 @@ import path = require('path');
 import azureStorage = require('azure-storage');
 import fs = require('fs');
 import models = require('artifact-engine/Models');
+import store = require('artifact-engine/Store');
 import tl = require('vsts-task-lib/task');
 
 export class AzureBlobProvider implements models.IArtifactProvider {
+
+    public artifactItemStore: store.ArtifactItemStore;
```
Comment (GitHubSriramB) [inline Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts]: where is this used?

File: Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts [hunk 2]
```diff
@@ -149,7 +149,7 @@ async function doWork() {
 
     } catch (err) {
         tl.debug(err.message);
```
Comment (GitHubSriramB) [inline Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts]: do we need to write this when we are writing the err next as error?

File: Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts [hunk 3]
```diff
@@ -149,7 +149,7 @@ async function doWork() {
...
-        tl._writeError(err);
+        tl.error(err);
```
Comment (GitHubSriramB) [inline Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts]: what is the difference between the two?

File: Tasks/DownloadBuildArtifacts/package.json [hunk 4]
```diff
@@ -17,8 +17,8 @@
   },
   "homepage": "https://github.com/Microsoft/vsts-tasks#readme",
   "dependencies": {
-    "vsts-task-lib": "0.9.20",
+    "vsts-task-lib": "2.1.0",
```
Comment (GitHubSriramB) [inline Tasks/DownloadBuildArtifacts/package.json]: why is this change reqd.?

File: Tasks/Common/azure-blobstorage-artifactProvider/package.json [hunk 5]
```diff
@@ -14,8 +14,8 @@
   "homepage": "https://github.com/Microsoft/vsts-tasks#readme",
   "dependencies": {
     "q": "1.4.1",
-    "artifact-engine": "0.1.4",
+    "artifact-engine": "0.1.10",
     "azure-storage": "2.2.1",
     "vsts-task-lib": "2.0.5"
```
Comment (GitHubSriramB) [inline Tasks/Common/azure-blobstorage-artifactProvider/package.json]: is it fine to use 2.0.5 here when we are using 2.1.0 in task?

File: Tasks/AzureVmssDeployment/package.json [hunk 6]
```diff
@@ -3,6 +3,6 @@
   "main": "main.js",
   "dependencies": {
     "vsts-task-lib": "2.0.5",
```
Comment (GitHubSriramB) [inline Tasks/AzureVmssDeployment/package.json]: should we use 2.1.0 here as well?

File: Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts (hunk 1 above)
Comment (omeshp) [reply Tasks/Common/azure-blobstorage-artifactProvider/azureBlobStorageProvider.ts]: not used currently, should be used to set downloadfilesize.

File: Tasks/DownloadBuildArtifacts/package.json (hunk 4 above)
Comment (omeshp) [reply Tasks/DownloadBuildArtifacts/package.json]: for task to work on 2.111.1 agent.

**Pull request merged**